### ISSUE TITLE
Inference mode problem

### ISSUE THREAD

**Issue by DaloroAT** [OML-Team/open-metric-learning#301]
An error occurs with torch=1.13.0 during metric calculation after using postprocessing. Probably, we should use a proper decorator for grad disabling

![image](https://user-images.githubusercontent.com/43877180/219854057-43c80d3c-1166-4816-b5b0-419c22b6d72f.png)

**Comment by DaloroAT** [comment]
Replacing decorator in `_inference` solves the problem

```python
# with torch.autocast(device_type="cuda", dtype=torch.float16 if use_fp16 else torch.float32):
#     with temporary_setting_model_mode(model, set_train=False):
with torch.no_grad():
```

**Comment by AlekseySh** [comment]
Is this a postprocessing specific problem?

**Comment by AlekseySh** [comment]
I installed `torch==1.13.0`, but cannot reproduce the error. 
I will try one more time later from my home computer

**Comment by AlekseySh** [comment]
I cannot reproduce the error on torch versions: 1.11, 1.12, 1.13.
So, I'm closing the issue.
